### PR TITLE
Add new render mode and template options for billboards

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This feature empowers admins to load raw HTML, like they can with `pages` via billboards, while still getting to leverage all the targeting and optimization functionality in billboards.

This creates new modes and templates, but could allow for additional in the future.

This will allow us to retire several less-developed features such as HTML variants and "Campaign sidebar" due to the flexibility this now provides admins.

This will be followed up by further UX adjustments to ensure admins are able to create and modify billboards in this way with confidence in not making mistakes — the initial version relies more heavily on admin scrutiny than future versions which should enable certain forms of encapsulation.

## Related Tickets & Documents

- Closes https://github.com/forem/forem/issues/20287

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [x] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

unlimited power

### CODE REVIEW

File: app/models/billboard.rb [hunk 1]
```diff
@@ -214,10 +216,11 @@ def generate_billboard_name
   def process_markdown
     return unless body_markdown_changed?
 
-    if FeatureFlag.enabled?(:consistent_rendering)
```
Comment (lightalloy) [inline app/models/billboard.rb:217]: I would like to remove the feature flag and use "consistent rendering" by default (for other models, like Articles).

Comment (benhalpern) [reply app/models/billboard.rb:217]: Yeah @lightalloy I was thinking the same thing, and having this be a part of that.

I made an issue for the broader task:

https://github.com/forem/forem/issues/20289

Labelled available for external, but grab it yourself if you're interested.

File: app/models/billboard.rb [hunk 2]
```diff
@@ -30,6 +30,8 @@ class Billboard < ApplicationRecord
   attribute :target_geolocations, :geolocation_array
   enum display_to: { all: 0, logged_in: 1, logged_out: 2 }, _prefix: true
   enum type_of: { in_house: 0, community: 1, external: 2 }
+  enum render_mode: { forem_markdown: 0, raw: 1 }
+  enum template: { authorship_box: 0, plain: 1 }
```
Comment (maestromac) [inline app/models/billboard.rb:34]: can you clarify what is meant here by template? and what is `authorship_box`?

Comment (benhalpern) [reply app/models/billboard.rb:34]: @maestromac `Template` is the wrapping HTML, so "authorship box" is the current template for billboards where we show the name/image of the org and/or the site name, etc.

"plain" is no visible template wrappe, but it still contains the div which stores the info to track events.

I used `template` to match the similar functionality we have with `pages` where we can either wrap the content in a certain context or just have it cover the whole page.

**Review by maestromac — APPROVED** [review]

**Pull request merged**